3.component.md

```diff
@@ -73,10 +73,10 @@ metadata:
   annotations:
     version: v1.0.0
     description: >
-      Sample component that describes the administrative interface for our Twitter bot.
+      Sample component that describes a very cool workload.
 spec:
-  type: Server
-  settings:
+  type: Server # <--- the type of the workload
+  settings: # <---- the settings of the workload
```

Comment:
This example seems to be different with current implementation, right?

Reply:
Yes, this change targets for OAM spec v0.2.2 release, and what we have implemented is OAM spec v0.2.1. The change has been synced with MSFT in maintainer meeting. We could sync internally if needed.

Reply:
The current crossplane implementation doesn't have the `type` and `settings` but I can see that the example format here is less K8s style and more generic.

Reply:
@ryanzhang-oss Yes, it's based on the discussion in maintainer call w/ MSFT folks, and it's easier to implement with compatibility than our previous discussion. OAM k8s runtime could implement it after spec v0.2.2 (i.e. the current working draft of this PR) is released.